src/eventHandlers/playerAttack.ts

```diff
@@ -0,0 +1,29 @@
+import type { ECSWorld } from '@/ecs/ECSWorld';
+import {
+  AnimatableBrand,
+  type Animatable,
+  AnimationState
+} from '@/entity/components/Animatable';
+import { PlayerBrand, type Player } from '@/entity/components/Player';
+import {
+  RenderableBrand,
+  type Renderable
+} from '@/entity/components/Renderable';
+import { objectAssign } from '@/utils/helpers';
+import type { Point } from '@/utils/types';
+
+export const playerAttackHandler = (mousePosition: Point, world: ECSWorld) => {
+  const [player] = world.entitiesByComponent<[Player, Animatable, Renderable]>([
+    PlayerBrand,
+    AnimatableBrand,
+    RenderableBrand
+  ]);
+
+  if (!player) return;
+
+  player.animatable.state = AnimationState.ATTACKING;
+  objectAssign(player.animatable.options, {
```

Comment:
We could make this a queue of animations, or a "next animation". With some animation types being interruptible by others. This would solve the animation cancelling we have. We could also not queue the same animation multiple times.